lib/src/analytics.dart

```diff
@@ -69,15 +81,35 @@ abstract class Analytics {
   /// [shouldShowMessage] returns true
   String get toolsMessage;
 
+  /// Returns a map representation of the [UserProperty] for the [Analytics] instance
+  ///
+  /// This is what will get sent to Google Analytics with every request
+  Map get userPropertyMap;
+
+  /// Call this method when the tool using this package is closed
+  ///
+  /// Prevents the tool from hanging when if there are still requests
+  /// that need to be sent off
+  void close();
+
+  /// API to send events to Google Analytics to track usage
+  void sendEvent({
+    required DashEvents eventName,
+    required Map eventData,
+  });
+
   /// Pass a boolean to either enable or disable telemetry and make
   /// the necessary changes in the persisted configuration file
   void setTelemetry(bool reportingBool);
 }
 
 class AnalyticsImpl implements Analytics {
   final FileSystem fs;
-  late ConfigHandler _configHandler;
+  late final ConfigHandler _configHandler;
   late bool _showMessage;
+  late final GAClient _gaClient;
+  late final String _clientId;
+  late final UserProperty userProperty;
```

Comment:
I left this one as not private because I thought it might be helpful for an IDE to have access to this possibly? It can provide visibility into what kind of user metadata we are sending. What do you think about exposing that kind of information?

Reply:
SGTM